programs/whirlpool/src/util/v2/remaining_accounts_utils.rs

```diff
@@ -0,0 +1,105 @@
+use anchor_lang::prelude::*;
+use crate::errors::ErrorCode;
+
+#[derive(AnchorSerialize, AnchorDeserialize, Clone, PartialEq, Eq)]
+pub enum AccountsType {
+    TransferHookA,
+    TransferHookB,
+    TransferHookReward,
+    TransferHookOneA,
+    TransferHookOneB,
+    TransferHookTwoA,
+    TransferHookTwoB,
+    //TickArray,
+    //TickArrayOne,
+    //TickArrayTwo,
+}
+
+#[derive(AnchorSerialize, AnchorDeserialize, Clone)]
+pub struct RemainingAccountsSlice {
+    pub accounts_type: AccountsType,
+    pub length: u8,
+}
+
+#[derive(AnchorSerialize, AnchorDeserialize, Clone)]
+pub struct RemainingAccountsInfo {
+    pub slices: Vec<RemainingAccountsSlice>,
+}
+
+#[derive(Default)]
+pub struct ParsedRemainingAccounts<'info> {
+    pub transfer_hook_a: Option<Vec<AccountInfo<'info>>>,
+    pub transfer_hook_b: Option<Vec<AccountInfo<'info>>>,
+    pub transfer_hook_reward: Option<Vec<AccountInfo<'info>>>,
+    pub transfer_hook_one_a: Option<Vec<AccountInfo<'info>>>,
+    pub transfer_hook_one_b: Option<Vec<AccountInfo<'info>>>,
+    pub transfer_hook_two_a: Option<Vec<AccountInfo<'info>>>,
```

Comment:
do we need these two? can two_hop_swap just reuse `transfer_hook_a/b`?

Reply:
I've merged "vault to vault transfer" implementation for two_hop_swap_v2.

two_hop_swap_v2 guarantees that there will be no excess of intermediate tokens.
note: It is simple implementation; If sqrtPriceLimit is used to generate a remainder of intermediate tokens, an error is generated.

As a result, an ATA for intermediate tokens is not needed.
So redundant inputs (one_a, one_b, two_a, two_b) have been improved and organized as input, intermediate and output.